src/GuppyDataExplorer/GuppyDataExplorer.jsx

```diff
@@ -13,29 +13,98 @@ import {
   ChartConfigType,
 } from './configTypeDef';
 import './GuppyDataExplorer.css';
+import isEnabled from '../helpers/featureFlags';
 
 class GuppyDataExplorer extends React.Component {
   constructor(props) {
     super(props);
+    let initialState = {};
+    let initialFilter = {};
+
+    // aborting URL decode
+    if (isEnabled('storeExplorerStateInURL')) {
+      const stateFromURL = getQueryParameter('filters');
+      if (stateFromURL) {
+        const decodedFilter = atob(stateFromURL);
+        const isValidJSON = IsValidJSONString(decodedFilter);
+        if (isValidJSON) {
+          initialState = JSON.parse(decodedFilter);
+          initialFilter = initialState.filter;
+        }
+      }
+    }
+
     this.state = {
       aggsData: {},
       filter: {},
+      initialFilterFromURL: initialFilter,
+      encodableExplorerStateForURL: { filter: initialFilter },
     };
   }
 
+  getEncodedQueryStateFromExplorer = () => {
+    const stateObj = this.state.encodableExplorerStateForURL;
+    if (this.isExplorerStatePristine(stateObj)) {
+      return '';
+    }
+    const encoded = encodeURIComponent(btoa(JSON.stringify(stateObj)));
+    return encoded;
+  }
+
+  refreshQueryStateInUrl = () => {
+    const encodedState = this.getEncodedQueryStateFromExplorer();
+    if (encodedState === '') {
+      window.history.pushState(null, null, window.location.pathname);
+      return;
+    }
+    window.history.pushState('', '', `?filters=${encodedState}`);
+  }
+
   handleReceiveNewAggsData = (newAggsData) => {
     this.setState({ aggsData: newAggsData });
   };
 
+  handleFilterChangeForQueryStateUrl = (eventIn) => {
+    let event = Object.assign({}, eventIn);
+    const newExplorerState = this.state.encodableExplorerStateForURL;
+    let field = Object.keys(event)[0];
+    for(let tabIndex = 0; tabIndex < this.props.filterConfig.tabs.length; tabIndex += 1) {
+      let searchFields = this.props.filterConfig.tabs[tabIndex].searchFields;
+      if(searchFields && searchFields.indexOf(field) !== -1) {
+        // Search fields are not supported in url-encoded explorer state at this time.
+        delete event[field];
+      }
+    }
+    newExplorerState.filter = event;
+    this.setState({ encodableExplorerStateForURL: newExplorerState });
+  }
+
+  isExplorerStatePristine = (explorerState) => {
+    const values = Object.values(explorerState);
+    for (let i = 0; i < values.length; i += 1) {
```

Comment:
minor: might use a `for...of` loop

Reply:
When I use a for...of loop like `for (const value of values) {`, i get this eslint error:

```
/Users/zakirgowani/Desktop/Projects/data-portal/src/GuppyDataExplorer/GuppyDataExplorer.jsx
  91:5  error  iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations  no-restricted-syntax
```

so i'm gonna leave it as is ~